modules/programs/gradle.nix

```diff
@@ -0,0 +1,106 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let cfg = config.programs.gradle;
+in {
+  options.programs.gradle = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+    };
+    package = mkOption {
+      type = types.package;
+      default = pkgs.gradle;
+      defaultText = literalExpression "pkgs.gradle";
+      example = "pkgs.gradle_7";
+      description = ''
+        The package to use for gradle.
+      '';
+    };
+    javaHome = mkOption {
+      type = types.nullOr types.package;
+      default = null;
+      example = "pkgs.jdk17";
+      description = ''
+        The Java package to use for running Gradle.
+
+        The default is not to use a dedicated Java package.
+        Instead programs.java.enable = true will be configured which
+        will result in JAVA_HOME being configured to that location.
+        Gradle will then pick up that Java package.
+
+        If a value is configured for this option it will be passed to
+        Gradle by setting the org.gradle.java.home property in
+        ~/.gradle/gradle.properties.
+      '';
+    };
+    javaInstallations = mkOption {
+      type = types.listOf types.package;
+      default = [ ];
+      example = "[ pkgs.jdk8 pkgs.jdk11 ]";
+      description = ''
+        Additional Java packages to make available to Gradle's Java toolchain infrastructure.
+        The configured packages will be made available by setting the org.gradle.java.installations.paths
+        property in ~/.gradle/gradle.properties.
+      '';
+    };
+    gradleProperties = mkOption {
+      type = types.attrsOf types.str;
```

Comment:
Would it be possible to use [RFC 0042-style](https://github.com/NixOS/rfcs/pull/42) settings here?

There is an example in NixOS’s [Apache Kafka module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/apache-kafka.nix#L37-L50).

Reply:
Done!